[19.0][MIG] payroll: migrate

tishmen opened the pull request:
Scope:

- payroll only

Depends on:

- None

Summary:

- Migrate to Odoo 19.
- Follows OCA 19.0 migration tasks (version bump, security field rename, tz API).

Pre-commit:

- Ran locally before opening; auto-fixes committed.

Tests:

- Local install example: `./odoo/odoo-bin -c odoo.conf -d odoo_payroll_dev -i payroll --log-level=info --stop-after-init`
- OCA CI will validate on upstream infra.

davejames committed:
payroll: rename module

this commit renames the module from hr_payroll to payroll. This is to avoid conflicting with the hr_payroll module included in Odoo Enterprise

davejames committed:
payroll: rename install setting for payroll_account module

Also removes links to localisations which have not yet been migrated. These can be re-added when the localisations are ported.

davejames committed:
payroll: add original pot file

davejames committed:
payroll: update new POT file

hilarak committed:
[IMP] payroll: black, isort, prettier

hilarak committed:
[MIG] payroll: Migration to 14.0

norlinhenrik committed:
[14.0][FIX] payroll: demo rule: python code with no indentation

norlinhenrik committed:
[14.0] payroll: new maintainer: appstogrow

nimarosa committed:
parent 830abfcae9440166206476c5b2c74932564483a1
author nicolasrsande <nicolasrsande@gmail.com> 1643135061 -0300
committer nicolasrsande <nicolasrsande@gmail.com> 1643668459 -0300

fix typo in leave calculation

default for worked days should not compute leaves, because leaves are calculated separately

we should not add data because it interfers with custom localization payroll modules

fix pre-commit

add demo data so the test can be executed

leaves should be computed in negative value to help creating salary rules

Add mantainer key

remove mantainer in payroll_account

14.0-minor-fixes

fix typo in leave calculation

we should not add data because it interfers with custom localization payroll modules

fix pre-commit

add demo data so the test can be executed

leaves should be computed in negative value to help creating salary rules

Add mantainer key

remove mantainer in payroll_account

14.0-payroll-minor-fixes

nimarosa committed:
add detail of the exeption in salary rule calculation

accept suggestion use repr() on the string

change indentation of error

nimarosa committed:
[14.0] [IMP] Code fields for python compute fields

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 93.6% (266 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/ca/

Noel000 committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/ca/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 86.9% (247 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (284 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

nimarosa committed:
[IMP] payroll: Add salary_rule_parameters functionality

fix build errors, the menu was loaded before the action

add menu to rule parameter

accept sugestions by appstogrow, will test in afternoon

fix typo

minor changes

[FIX] payroll: remove payslip view change. Will make it in another PR

nimarosa committed:
[IMP] payroll: take into account the contract start date in payslip calculation

nimarosa committed:
[IMP] payroll: recalculate worked_days and inputs if structure is changed

nimarosa committed:
[IMP] payroll: add amount_qty to paylip inputs

nimarosa committed:
[IMP] payroll: add missing states in status bar

nimarosa committed:
[IMP] payroll: ux changes in salary rules views

Commit:
[ADD] payroll: missing tag to compute method

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 78.1% (222 of 284 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/it/

nimarosa committed:
[IMP] payroll: add support for hr_public_holidays in work time calculation

nimarosa committed:
[IMP] payroll: remove empty lines

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[FIX] payroll unittest

The unittest for the payroll module is broken in several ways:
    1. It doesn't actually test anything in the payslip, just the states
       when various buttons are simulated
    2. The payslip is created programatically, so no payslip values are loaded
    3. The employee's contract is in "draft" state so even if it tried to
       load any values it wouldn't work because the contract isn't "open"
    4. It modifies a salary input line, but doesn't verify it worked. It
       doesn't work because of the above mentioned reasons.

To ensure a minimal amount of testing:
    o Make sure the employee's contract is in "open" state
    o Use the "Form" testing object when creating a payslip to ensure
      the onchange_employee() is run
    o Actually test the payroll calculations are done correctly

mtelahun committed:
[FIX] payroll - wrong quantity and rate during python computation of salary rule

A side effect of the way the payslip line quantity and rate are set when
amount_select is set to "code" causes them to never evaluate to zero. When they
evaluate to 0 the side effect causes them to be set to 1.0 and 100.0, respectively.

    "result_qty" in localdict and localdict["result_qty"] or 1.0
    "result_rate" in localdict and localdict["result_rate"] or 100.0

mtelahun committed:
[IMP] payroll: refactor of local dictionary handling in payslip for salary rule processing

This should make it easier for other modules to inherit and modify the
payroll module. This solution allows other modules to arbitrarily
inject their own values into salary rules. Furthermore, payroll managers
will be able to continue using '.' (dot) notation to refer to these new
values. Module implementers can inherit and overload the following methods:

get_payroll_dict()  - for providing values under the generic `payroll.` object
get_baselocaldict() - for providing 'first-level' objects that are
                      available directly to salary rules. For the sake
                      of clarity the object string should start with
                      the module name. Example: if the module is named
                      payroll_foo the values should be under `payroll_foo.`
get_contract_dict() - for values that depend on the current contract being evaluated
                      These values will be available under `this_contract.`

In the inherited methods the modules should return a python dictionary of
discreet values and/or nested objects using a BaseBrowsableObject.
Example:
If the module name is my_module:
    res = super().get_baselocaldict(contract, contracts)
    res.update({
        "my_module": BaseBrowsableObject(
            "foo": 10.0,
            "bar": 25.0,
            "baz": BaseBrowsableObject(
                {
                    value1: 100.0,
                    value2: 150.0,
                }
            )
        )
    })
    return res

The above values can be accessed in salary rules as:
my_module.foo
my_module.bar
my_module.baz.value1
my_module.baz.value2

mtelahun committed:
[IMP] payroll: remove dependency on demo data

mtelahun committed:
[IMP] payroll: add a test to show that if result_* are not set they get default values

nimarosa committed:
[IMP] payroll: Remove contract_advantage_template model and new module hr_contract_advantages

[ADD] Add new module hr_contract_advantages

[FIX] payroll_contract_advantages: Fix typo in view

[FIX] payroll_contract_advantages: Fix typo in import

[IMP] payroll: change security grups

[FIX] payroll: fix version

[FIX] payroll_contract_advantages: fix import typo

[FIX] payroll_contract_advantages: Fix security groups

[FIX] payroll_contract_advantages: Fix menu typo

[IMP] payroll_contract_advantages: Fix get_contract_dict function

nimarosa committed:
[IMP] payroll: payslip refactoring and functionality improvements

[IMP] payroll: refactor of the method get_worked_day_lines

[IMP] payroll: remove old implementation

[IMP] payroll: remove unused lines

[IMP] payroll: improve payslip views

[IMP] payroll: Add input line computation in onchange

[IMP] payroll: Add a button to refetch payslip data manually

[IMP] payroll: Add domain to contract in many2one tables

[IMP] payroll: Add more tracking in fields

[IMP] payroll: extracted the name computing method

[FIX] payroll: remove redundant method

[IMP] payroll: replace tracking=1 with tracking=True

[IMP] payroll: Add daterange widget for dates

[IMP] payroll: Refactor get_payslip_vals and add struct_id selection on runs

[IMP] payroll: In runs if struct_id is not selected we fallback to contract default

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (290 of 290 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (301 of 301 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[IMP] payroll: allow the use of positive values in leave days/hours calculation

The system parameter "payroll.leaves_positive" is used to control
whether the value for number_of_days/number_of_hours is positive or
negative. When False (the default) it wil be negative.

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (307 of 307 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (307 of 307 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

norlinhenrik committed:
[IMP] payroll: send email with payslip, fix payslip form view, better payslip line views

nimarosa committed:
[FIX] payroll: Fix Refactored _compute_name to contemplate more than one record

[FIX] payroll: fix _compute_name singleton error

mtelahun committed:
[FIX] payroll: use a custom working calendar so tests don't fail on weekends

nimarosa committed:
[IMP] payroll: Improve views, add child functionalities, allow computing name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[FIX] payroll: fix test expected values

[IMP] payroll: [IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: Add test coverage for new child view and flag

[IMP] payroll: Improve views, add child functionalities, allow compute of name

[IMP] payroll: change tab name

[IMP] payroll: Add info alert in salary_rule form explaining functionality

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (304 of 304 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (304 of 304 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (313 of 313 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (313 of 313 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
[IMP] payroll: hr_payslip refactoring and misc improvements

[IMP] payroll: Move and refactor category sum function (now can be inherited)

[IMP] payroll: _get_payslip_lines refactoring

[IMP] payroll: move basebrowsableobject to own file

[IMP] payroll: adapt compute method to new returned values

[IMP] payroll: Add compute_date and use "verify" unused state - improve views and filters

[IMP] payroll: Add compute_date and use "verify" unused state - improve views and filters

[IMP] payroll: misc changes

[FIX] payslip_change_state: Fix tests and exeptions acording new implementation

[ADD] hr_payroll_cancel: Add mantainer keys

[IMP] payroll: Change more detailes name for lines

[IMP] all: Change repo description and contributors file.

[IMP] payroll: change order

[IMP] payroll: minor changes

[IMP] payroll: minor changes

norlinhenrik committed:
[IMP] payroll: hr.payslip.line with date_from

norlinhenrik committed:
[IMP] payroll: hr.payslip & hr.payslip.run: order by id desc (most recently created on top)

mtelahun committed:
[IMP] payroll: default value for python code fields had lines in the comments that were not commented

mtelahun committed:
[IMP] payroll: make hr.salary.rule (form) condition and computation views more sane in Firefox

nimarosa committed:
[IMP] payroll: Add hook method for accesing localdict

[IMP] payroll: Add hook method for accesing localdict

nimarosa committed:
[IMP] payroll: extract and move _sum_salary_rule_category to payslip

nimarosa committed:
[IMP] payroll: add new objects documentation in help text

nimarosa committed:
[IMP] payroll: remove ** from baselocadict

nimarosa committed:
[IMP] payroll: remove sum from worked days table

nimarosa committed:
[IMP] payroll: Add object for accessing with dot notation to the dict

nimarosa committed:
[IMP] payroll: Merge payslip_cancel in base module and remove module

nimarosa committed:
[IMP] payroll: Merge payslip_change_state and remove module

nimarosa committed:
[IMP] payroll: add tests for new merged functionalities

[FIX] payroll: fix tests

[FIX] payroll: fix tests

[IMP] payroll: Add conditional button and computed method for setting

[IMP] payroll: Add conditional button and computed method for setting

[IMP] payroll: Move cancel test to payroll account

[IMP] payroll: Add hr_payslio_cancel refund sheet code

[FIX] payroll: tests change import

[FIX] payroll: Remove account in tests, will add another test in account module

[IMP] payroll_account: Add cancel tests for accounting module

[FIX] payroll_account: Test remove

[FIX] account_payroll: Fix tests

nimarosa committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (315 of 315 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (315 of 315 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es/

nimarosa committed:
[IMP] payroll: change button attrs

mtelahun committed:
payroll: silence pylint W7950(odoo-addons-relative-import)

mtelahun committed:
[FIX] payroll: regression introduced in #62

When calling _compute_payslip_line() call it on the payslip object and
not recordset 'self'.

mtelahun committed:
[IMP] payroll: refactor tests.common to prepare for more tests

mtelahun committed:
[IMP] payroll: re-factor _get_contracts() to work on recordset self

mtelahun committed:
[IMP] payroll: re-factor _get_salary_rules() to work on recordset self

mtelahun committed:
[IMP] payroll: rename and re-factor _get_payslip_lines() to work on recordset self

To preserve compatibility with third-party modules the re-factored
functionality is in a new method: get_lines_dict() and the
_get_payslip_lines() just calls this method. I expect that the new
multi-recordset function will be useful in reports: no more manually
adding up payslip lines from individual payslips!

mtelahun committed:
[IMP] payroll: re-factor onchange_struct_id() to work on recordset self

mtelahun committed:
[IMP] payroll: re-factor onchange_dates() and onchange_employee() to work on recordset self

mtelahun committed:
payroll: fix a couple of places where functional changes were inadvertently introduced during the refactor

norlinhenrik committed:
[IMP] payroll: rule _satisfy_condition() & _compute_rule() -> 2*4 methods

norlinhenrik committed:
[IMP] payroll: rule._compute_rule() returns dict of values

norlinhenrik committed:
[IMP] payroll: payslip.get_lines_dict() calls rule._reset_localdict_values()

norlinhenrik committed:
[IMP] payroll: payslip._compute_payslip_line() -> _compute_rule() & _get_line_dict()

norlinhenrik committed:
[IMP] payroll: rule._compute_rule_code() calls _get_rule_dict()

nimarosa committed:
[FIX] payroll_contract_advantages: Function name changed in recent commits

norlinhenrik committed:
[IMP] payroll: payslip._compute_payslip_line() & _get_lines_dict()

nimarosa committed:
[14.0] [FIX] payroll: Fix get_lines_dict() - skip assigning dict when the rule is blacklisted

nimarosa committed:
[14.0] [IMP] payroll: support adding python libraries or objects to localdict

nimarosa committed:
[14.0] [IMP] payroll: improve lines table view

nimarosa committed:
[14.0] [IMP] payroll: improve salary_rule and categories views and documentation

[14.0] [IMP] payroll: improve salary rules views and documentation

[14.0] [IMP] payroll: improve salary rules  and categories views

[14.0] [IMP] payroll: improve salary rules views and documentation

[14.0] [IMP] payroll: improve salary rules views and documentation

[IMP] payroll: fix repeated words

[14.0] [IMP] payroll: fix typos

nimarosa committed:
[14.0] [IMP] payroll: settings for payslip_lines edition and auto-compute on confirm

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

[14.0] [IMP] payroll: Add no compute on confirm setting

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 99.7% (363 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (364 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

nimarosa committed:
[14.0] [FIX] payroll: Fix parent rules dependecy

nimarosa committed:
[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [IMP] payroll: Add rule identification on error

[14.0] [FIX] payroll: Add tests about parent/child rules

[14.0] [IMP] payroll: Rollback rule identification on error

[14.0] [FIX] payroll: Add tests about parent/child rules

nimarosa committed:
[14.0] [ADD] hr_payroll_period

[14.0] [ADD] hr_payroll_period: Add mantainers key

[14.0] [ADD] hr_payroll_period: Change manifest category

[14.0] [ADD] hr_payroll_period

[14.0] [ADD] hr_payroll_period

[14.0] [IMP]  hr_payroll_period: Improve views

[14.0] [IMP] hr_payroll_period: Improve views and misc

[14.0] [IMP] hr_payroll_period: Improve views and misc

[14.0] [IMP] hr_payroll_period: Improve views and misc

nimarosa committed:
[14.0] [IMP] payroll: Add employee in rule exceptions

[14.0] [IMP] payroll: Add employee in rule exceptions

norlinhenrik committed:
[IMP] payroll: hr.salary.rule is readonly for payroll user

norlinhenrik committed:
[IMP] payroll: depends on base_time_parameter

[FIX] payroll: rule_parameter

[MIG] payroll: depends on base_time_parameter

[FIX] payroll migration

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (364 of 364 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

norlinhenrik committed:
[ADD] payroll_rule_time_parameter (instead of payroll) depends on base_time_parameter

norlinhenrik committed:
[IMP] payroll: history

norlinhenrik committed:
[IMP] payroll: hr.payslip.run with company_id

[FIX] payroll: fix views

norlinhenrik committed:
[IMP] payroll: code & category are not required

norlinhenrik committed:
[IMP] payroll: settings descriptions, show_details_by_salary_rule_category

norlinhenrik committed:
[IMP] payroll: required fields, removed details_by_salary_rule_category

norlinhenrik committed:
[IMP] payroll: hide_invisible_lines (who don't appear on payslip)

norlinhenrik committed:
[IMP] payroll: New setting: Require code and category

norlinhenrik committed:
[IMP] payroll: test rule code in payslip line

nimarosa committed:
[IMP] payroll: remove parameters code

nimarosa committed:
[14.0] [IMP] payroll: add localdict pure payslip object and rename payslips browsable object

nimarosa committed:
[14.0] [IMP] payroll: remove old tests from payroll module

nimarosa committed:
[14.0] [IMP] payroll: improve comments and documentation of available objects

nimarosa committed:
[14.0] [IMP] payroll: change manifest category

[14.0] [IMP] payroll: change manifest category

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

[14.0] [IMP] payroll: add migration for new payslip and payslips objects

norlinhenrik committed:
[IMP] payroll: removed required=False, improved text descriptions

norlinhenrik committed:
[IMP] payroll: test rule and category with and without code

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (359 of 359 strings)

Translation: payroll-14.0/payroll-14.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-14-0/payroll-14-0-payroll/es_AR/

mtelahun committed:
[MIG] payroll: Migration to 15.0

mtelahun committed:
[MIG] payroll: Migration to 16.0

Commit:
[UPD] Update payroll.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

victoralmau committed:
[FIX] payroll: Change mail template expressions

victoralmau committed:
[FIX] payroll: Fix tests since https://github.com/odoo/odoo/commit/09091dded0115c3af1e8730885fa2f018d17d128

Now a contract is closed whenever appropriate, therefore, the expected
value must be changed in tests.

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

OCA-git-bot committed:
[UPD] README.rst

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

mostafabarmshory committed:
Translated using Weblate (Persian)

Currently translated at 98.8% (354 of 358 strings)

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/fa/

mostafabarmshory committed:
Added translation using Weblate (Persian (fa_IR))

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (358 of 358 strings)

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/es/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 57.2% (205 of 358 strings)

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/it/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-16.0/payroll-16.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-16-0/payroll-16-0-payroll/

FernandoRomera committed:
[17.0][MIG] payroll: Migration to 17.0

cvinh committed:
[FIX] Fix tests in payroll

hapolinario committed:
[FIX] struct_id force_save and test_get_contracts_singleton

hapolinario committed:
[FIX] pre-commit ruff

hapolinario committed:
[FIX] pre-commit ruff

hapolinario committed:
[FIX] field very narrow and show duplicate

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-17.0/payroll-17.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-17-0/payroll-17-0-payroll/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (354 of 354 strings)

Translation: payroll-17.0/payroll-17.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-17-0/payroll-17-0-payroll/it/

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (354 of 354 strings)

Translation: payroll-17.0/payroll-17.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-17-0/payroll-17-0-payroll/sv/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (354 of 354 strings)

Translation: payroll-17.0/payroll-17.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-17-0/payroll-17-0-payroll/it/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-17.0/payroll-17.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-17-0/payroll-17-0-payroll/

victoralmau committed:
[FIX] payroll: Remove migrations folder from 16.0

regispirard committed:
[MIG] payroll: Migration to 18.0

regispirard committed:
[IMP] payroll: pre-commit execution

dreispt committed:
[FIX] payroll: labels missing for salary info fields

dreispt committed:
[FIX] payroll: do not indent default code comments

Commit:
[UPD] Update payroll.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (356 of 356 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/it/

dreispt committed:
[IMP] payroll: UX improvements, better error messages

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/it/

dreispt committed:
[FIX] payroll: error testing report printing

dreispt committed:
[REF] payroll: use recordsets instead int lists

victoralmau committed:
[FIX] payroll: Display Salary Structure form correctly

TT54568

victoralmau committed:
[IMP] payroll: Improve Payslip form view

Define as h1 the field name instead of the field employee_id

TT54568

victoralmau committed:
[IMP] payroll: Purge translations

Commit:
[UPD] Update payroll.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/

OCA-git-bot committed:
[BOT] post-merge updates

dreispt committed:
[FIX] payroll: test could set contract end before start date

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es/

macagua committed:
Translated using Weblate (Spanish (Venezuela))

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es_VE/

macagua committed:
Translated using Weblate (Spanish (Venezuela))

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es_VE/

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es/

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es/

macagua committed:
Translated using Weblate (Spanish (Venezuela))

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es_VE/

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es/

macagua committed:
Translated using Weblate (Spanish (Venezuela))

Currently translated at 100.0% (357 of 357 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/es_VE/

OCA-git-bot committed:
[BOT] post-merge updates

nimarosa committed:
[18.0][FIX] payroll: Correct leaves and fix view

nimarosa committed:
[18.0][FIX] payroll: Fix tests to reflect new changes

nimarosa committed:
[18.0][FIX] payroll: Check if leave has work_entry_type defined or fallback to default

nimarosa committed:
[18.0][FIX] payroll: Check if leave has work_entry_type defined or fallback to default

Commit:
[UPD] Update payroll.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (355 of 355 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/it/

Commit:
[UPD] Update payroll.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/

tsezgin committed:
Translated using Weblate (Turkish)

Currently translated at 59.8% (212 of 354 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/tr/

tsezgin committed:
Translated using Weblate (Turkish)

Currently translated at 82.7% (293 of 354 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/tr/

tsezgin committed:
Translated using Weblate (Turkish)

Currently translated at 95.7% (339 of 354 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/tr/

tsezgin committed:
Translated using Weblate (Turkish)

Currently translated at 100.0% (354 of 354 strings)

Translation: payroll-18.0/payroll-18.0-payroll
Translate-URL: https://translation.odoo-community.org/projects/payroll-18-0/payroll-18-0-payroll/tr/

Commit:
[IMP] payroll: pre-commit auto fixes

tishmen commented:
- Translation best practice in Odoo is to keep placeholders inside the translatable string and interpolate after translation. That pattern looks like `self.env._("… %(name)s …") % {"name": value}`.
- pylint-odoo ports a logging “not‑lazy” rule to translations and flags this interpolation as `W8301`, but there is no “lazy‑args” form for `_()`, so the warning is a stylistic false positive for Odoo’s i18n pattern. To remain translator‑friendly and still pass the hook, I added precise, per‑line disables only on the exact interpolation lines (`# pylint: disable=translation-not-lazy`).
- Ruff’s `UP031` prefers `.format`/f‑strings over `%` formatting. For translatable strings, the percent style with named placeholders is a better fit for Odoo’s extraction and translation workflows. I therefore added per‑line `# noqa: UP031` where interpolation occurs after translation to avoid fighting the formatter and to stay consistent with OCA modules.

Commit:
[MIG] payroll: Migration to 19.0

[REF] payroll: remove explicit 'hr' dependency

[FIX] payroll: fix demo data and review nits

- Drop hr.contract demo records (Odoo 19)

- Use calendar/employee timezone for leave computation

- Replace translation % formatting with env._ kwargs

- Remove redundant payslip unlink override

Show salary structure on employee contracts

- Add struct_id to the Employee/Payroll contract overview
- Copy struct_id from contract templates (whitelist)

Functional

tishmen commented:
- Odoo 19 drops the separate hr_contract module; “contracts” are now timeline versions inside core HR via the hr.version model.
hr.employee _inherits hr.version (current version in employee.version_id), with multiple versions capturing effective changes via date_version + contract_date_start/contract_date_end.
- I replaced hr.contract usages with hr.version: fields like date_start/date_end → date_version/contract_date_start/contract_date_end, and selection via employee._get_contracts(date_start, date_end), considering only overlapping versions.
- Views/security were adjusted to target hr.version (contract templates), while hr.contract.type remains in hr for employment type metadata.
- To preserve payroll features, I extend hr.version (not hr.contract) to carry payroll-specific fields (e.g., struct_id, schedule_pay) and keep existing flows working on the new model.

tishmen commented:
@kaynnan @WesleyOliveira98 @marcelsavegnago @CristianoMafraJunior pre-commit and tests are green; please test on RunBoat and review/approve.

kaynnan commented on payroll/__manifest__.py:
```diff
@@ -2,15 +2,15 @@
     "author": "Odoo SA, Odoo Community Association (OCA)",
     "depends": [
-        "hr_contract",
+        "hr",
```
Is this HR import really necessary? Because hr_holidays already has HR in its manifest.

tishmen replied:
Resolved in: https://github.com/OCA/payroll/pull/227/commits/0695c91085567f43cd70704a7c7e3e81a1bdfd76

CristianoMafraJunior reviewed (changes requested):
The commit message should be only: [MIG] payroll: Migration to 19.0
970eaf703780067e83271b11bff4102728bef870

tishmen commented:
> The commit message should be only: [MIG] payroll: Migration to 19.0 [970eaf7](https://github.com/OCA/payroll/commit/970eaf703780067e83271b11bff4102728bef870)

I have changed the commit message to the one specified, thank you.

dreispt commented on payroll/models/hr_payslip.py:
```diff
@@ -268,11 +270,16 @@ def refund_sheet(self):
         return res
 
     def unlink(self):
+        return super().unlink()
```
This could be just removed, right?

dreispt commented on payroll/models/hr_payslip.py:
```diff
@@ -333,7 +340,7 @@ def _compute_leave_days(self, contract, day_from, day_to):
         leaves = {}
         calendar = contract.resource_calendar_id
-        tz = timezone(calendar.tz)
+        tz = self.env.tz
```
No, you want the Employees tz, not te users's

dreispt commented on payroll/models/hr_payslip.py:
```diff
@@ -757,10 +786,10 @@ def _compute_name(self):
                 locale=record.env.context.get("lang") or "en_US",
             )
-            record.name = _("Salary Slip of %(name)s for %(dt)s") % {
+            record.name = self.env._("Salary Slip of %(name)s for %(dt)s") % {  # pylint: disable=translation-not-lazy
```
This looks like a translatable string.

dreispt commented on payroll/models/hr_payslip.py:
```diff
@@ -216,22 +215,25 @@ def action_payslip_cancel(self):
-                {"credit_note": True, "name": _("Refund: %s") % payslip.name}
+                {
+                    "credit_note": True,
+                    "name": (self.env._("Refund: %(name)s") % {"name": payslip.name}),  # noqa: UP031  # pylint: disable=translation-not-lazy
```
This looks like a translatable string.

dreispt commented on payroll/models/hr_payroll_structure.py:
```diff
@@ -60,12 +62,16 @@ def _compute_require_code(self):
-        default = dict(default or {}, code=_("%s (copy)") % self.code)
+        default = dict(
+            default or {},
+            code=(self.env._("%(code)s (copy)") % {"code": self.code}),  # noqa: UP031  # pylint: disable=translation-not-lazy
```
Can the noqa comment be removed?

dreispt commented on payroll/models/hr_payslip.py:
```diff
@@ -757,10 +786,10 @@ def _compute_name(self):
                 "name": record.employee_id.name,
                 "dt": str(date_formatted),
-            }
+            }  # noqa: UP031
```
Why is this needed?

dreispt commented on payroll/models/hr_payslip_line.py:
```diff
@@ -61,10 +61,14 @@ def _compute_parent_line_id(self):
+                            "%(rule)s"
+                        )
+                        % {"rule": line.parent_rule_id.name}  # noqa: UP031  # pylint: disable=translation-not-lazy
                     )
```
Again, I have concerns with the noqa comments, and feel they are not really required

dreispt commented on payroll/models/hr_salary_rule.py:
```diff
@@ -210,7 +212,8 @@ def _compute_rule(self, localdict):
         self.ensure_one()
         method = f"_compute_rule_{self.amount_select}"
-        return api.call_kw(self, method, [self.ids, localdict], {})
+        compute = getattr(self, method)
```
Yes!

dreispt reviewed (commented):
Thank you, this looks like solid and brave work.
I made a few minor comments.
The one that could be a bug is the tz handling.

realsaiko commented:
Demo data installation fails:
`./odoo/odoo-bin -c odoo.conf -d odoo_payroll_dev -i payroll --with-demo --log-level=info --stop-after-init`
It still has a reference to the "hr.contract" model that was removed in Odoo v19.0

tishmen commented:
Thanks for the detailed review. I’ve pushed updates addressing your points: removed the redundant unlink() override, fixed leave-day timezone handling to use the contract/calendar timezone, and refactored the translation strings to use env._(..., **kwargs) so we can drop the noqa/pylint suppressions. Please take another look when you have a moment.

tishmen commented:
@realsaiko The demo data still referenced hr.contract, which no longer exists in Odoo 19. I’ve updated the demo XML to remove those hr.contract records and instead set the contract/version fields on the employee records, so payroll --with-demo should install cleanly now. If you can retry the install and confirm, that’d be very helpful.

realsaiko commented:
@tishmen great, now the demo installation works without errors

hieulucky111 commented:
I don't see where we can set the Salary Structure on the Employee Version (Contract)
<img width="1274" height="816" alt="image" src="https://github.com/user-attachments/assets/c89cce6c-e10d-48e0-bbe7-e7ad036f2cab" />

hieulucky111 commented:
The rest seems ok for me. Functional Test OK

tishmen commented:
<img width="749" height="921" alt="image" src="https://github.com/user-attachments/assets/8f5a9aae-4714-4293-bb86-a42c8dd5003c" />

@hieulucky111 In Odoo 19 the “contract” is an hr.version edited through the Employee form (the “New Contract” button creates a new version and reloads the employee with context['version_id']), so adding struct_id only on the contract template form
  wasn’t enough. I added Salary Structure (struct_id) to the Employee → Payroll → Contract Overview and included it in the template whitelist so “Load a Template” can set it too. Verified on runboat oca-payroll-19-0-pr227-af64a2c7fb96: the field now renders right after “Pay Category”.

hieulucky111 reviewed (approved):
Functional Test Ok

dalonsod committed:
[FIX] payroll: make again payslip name editable at draft state

Prior to v17 that was default behavior, this restores it.

(cherry picked from commit 1a2e79458d1039167a3d4c913fe8a5a7d401a6f8)

dalonsod commented:
Hello @tishmen could cherry-pick #248 ?

tishmen commented:
Hi @dalonsod, I cherry-picked #248 into this migration PR. The fix is now included on 19.0-mig-payroll. Thanks!

tishmen commented:
Hi all, I squashed the follow-up commits into a single migration commit, keeping the pre-commit auto-fixes commit intact and leaving the cherry-picked fix as a separate commit on top. CI is green again.

BhaveshHeliconia reviewed (approved):
Code and functional review LGTM!

Yusuke1998 commented:
ping

marcelsavegnago reviewed (approved):
LGTM

dreispt commented:
/ocabot merge nobump
